### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by psheu** [csoulakian/cs330s16p1#5]
I really appreciate that everything is commented well. I could follow your code with ease and understand how you went about the different methods. Having the separate test fixtures class seems to add complexity, though it allows testing to be very thorough. It might help to just include in the ParserTest several valid and invalid tweets that include hashtags, mentions and URLs.

**Comment by csoulakian** [comment]
Thank you for your feedback! 
The Parser only takes in a string array and I have already filtered out messages that are too long or empty in the Reader. I'm not sure when you checked out the code, but last night I added 3 more tests of messages that include a mention that is too long so it gets thrown out when parsed and a hashtag with an invalid character in it so it gets shortened to cut off where the invalid character starts.

`static String[] L18 = {"#this.project", "was", "#so..fun,", "@Billy"};`
`static String[] L19 = {"#thatawkwardmomentwhen", "your", "@lastnameistoolong"};`
`static String[] L20 = {"google.com", "@google.com", "#google.com"};`

Is that what you were referring to? I'll also make sure to include an invalid URL.
